packages/odyssey-react-mui/src/TextField.tsx

```diff
@@ -152,91 +129,56 @@ const TextField = forwardRef<HTMLInputElement, TextFieldProps>(
-    const labelId = label ? `${id}-label` : undefined;
-
-    const localInputProps = useMemo(() => {
-      const ariaDescribedBy =
-        errorId && hintId ? `${hintId} ${errorId}` : errorId || hintId;
-
-      return {
-        ...inputProps,
-        "aria-describedby":
-          inputProps["aria-describedby"]?.concat(` ${ariaDescribedBy}`) ??
-          ariaDescribedBy,
-      };
-    }, [errorId, hintId, inputProps]);
-
-    return (
-      <FormControl disabled={isDisabled} error={Boolean(errorMessage)}>
-        <InputLabel htmlFor={id} id={labelId}>
-          {label}
-          {!isRequired && (
-            <Typography variant="subtitle1">{optionalLabel}</Typography>
-          )}
-        </InputLabel>
-        {hint && <FormHelperText id={hintId}>{hint}</FormHelperText>}
+    const renderFieldComponent = useCallback(
+      ({ ariaDescribedBy, id }) => (
         <InputBase
+          aria-describedby={ariaDescribedBy}
           autoComplete={autoCompleteType}
           /* eslint-disable-next-line jsx-a11y/no-autofocus */
           autoFocus={autoFocus}
-          endAdornment={
-            type === "password" ? (
-              <InputAdornment position="end">
-                <IconButton
-                  aria-label="toggle password visibility"
-                  edge="end"
-                  onClick={togglePasswordVisibility}
-                >
-                  {inputType === "password" ? <EyeIcon /> : <EyeOffIcon />}
-                </IconButton>
-              </InputAdornment>
-            ) : (
-              endAdornment
-            )
-          }
+          endAdornment={endAdornment}
           id={id}
-          inputProps={localInputProps}
           multiline={isMultiline}
+          onBlur={onBlur}
           onChange={onChange}
           onFocus={onFocus}
-          onBlur={onBlur}
           placeholder={placeholder}
           readOnly={isReadOnly}
           ref={ref}
-          startAdornment={
-            inputType === "search" ? (
-              <InputAdornment position="start">
-                <SearchIcon />
-              </InputAdornment>
-            ) : (
-              startAdornment
-            )
-          }
-          type={inputType}
+          startAdornment={startAdornment}
+          type={type}
           value={value}
         />
-        {errorMessage && (
-          <FormHelperText error id={errorId}>
-            <span style={visuallyHidden}>Error:</span>
-            {errorMessage}
-          </FormHelperText>
-        )}
-      </FormControl>
+      ),
+      [
+        autoCompleteType,
+        autoFocus,
+        endAdornment,
+        isMultiline,
+        onChange,
+        onFocus,
+        onBlur,
+        placeholder,
+        isReadOnly,
+        ref,
+        startAdornment,
+        type,
+        value,
+      ]
+    );
+
+    return (
+      <Field
+        errorMessage={errorMessage}
+        hasVisibleLabel
```

Comment:
Maybe this should be optional if we have it default to `true`? Not strongly opinionated there, since we want visible labels as a standard.

Reply:
This is like `hint` and `error`. We're defining what we want on `Field`. Only 6-7 components will use it. Even if we expose it, this is all for dev maintenance. Someone can look at it and know what to expect. Rather than telling them to turn off a feature, they're telling this component what they want to show up.